File: app/src/main/java/tm/nsfantom/a2048k/ui/SimpleArrayAdapter.java
package tm.nsfantom.a2048k.ui;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import tm.nsfantom.a2048k.R;
import tm.nsfantom.a2048k.model.StatItem;

/**
 * Created by user on 2/15/18.
 */

public class SimpleArrayAdapter extends ArrayAdapter<StatItem> {
    ArrayList<StatItem> statItems;
    Context context;

    public SimpleArrayAdapter(@NonNull Context context, @NonNull ArrayList<StatItem> objects) {
        super(context, -1, objects);
        this.statItems = objects;
        this.context = context;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View rowView = inflater.inflate(R.layout.item_stats, parent, false);
        TextView resultView = (TextView) rowView.findViewById(R.id.tvResult);
        TextView dateView = (TextView) rowView.findViewById(R.id.tvDate);
        TextView positionView = (TextView) rowView.findViewById(R.id.tvPosition);
        StatItem statItem = statItems.get(position);
        resultView.setText(String.valueOf(statItem.getResult()));
        // change the icon for Windows and iPhone
        dateView.setText(String.valueOf(statItem.getDate()));
        positionView.setText(String.valueOf(position+1));

        return rowView;
    }
}

File: app/src/main/java/tm/nsfantom/a2048k/ui/SimpleArrayAdapter.kt
package tm.nsfantom.a2048k.ui

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.item_stats.view.*
import tm.nsfantom.a2048k.R
import tm.nsfantom.a2048k.model.StatItem
import java.util.*
import android.text.format.DateUtils
import java.text.SimpleDateFormat


class SimpleArrayAdapter(internal var context: Context, internal var statItems: ArrayList<StatItem>) : ArrayAdapter<StatItem>(context, -1, statItems) {

    fun updateData(statItems: ArrayList<StatItem>){
        this.statItems = statItems
        notifyDataSetChanged()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val inflater = parent.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val rowView = inflater.inflate(R.layout.item_stats, parent, false)
        val statItem = statItems[position]
        rowView.tvResult.text = context.getString(R.string.score_label,statItem.result)
        rowView.tvDate.text = formateDate(statItem.date)//statItem.date.toString()
        rowView.tvPosition.text = statItem.rank.toString()
        return rowView
    }

    private fun formateDate(millis: Long):String{
        val sdf = SimpleDateFormat("dd-MM-yyyy\nHH:mm")
        sdf.setTimeZone(TimeZone.getTimeZone("GMT"))
        val now = System.currentTimeMillis()
        if ((now-millis)/1000 > 60*60*24*2)
            return sdf.format(Date(millis))
        return DateUtils.getRelativeTimeSpanString(millis, now, DateUtils.MINUTE_IN_MILLIS).toString()
    }
}
